src/gateway/handlers/messageCreate.ts

```diff
@@ -0,0 +1,20 @@
+import { Channel } from "../../structures/channel.ts"
+import { Message } from "../../structures/message.ts"
+import { MessageMentions } from "../../structures/MessageMentions.ts"
+import { User } from "../../structures/user.ts"
+import { MessagePayload } from "../../types/channelTypes.ts"
+import { Gateway, GatewayEventHandler } from '../index.ts'
+
+export const messageCreate: GatewayEventHandler = async(
+  gateway: Gateway,
+  d: MessagePayload
+) => {
+  let channel = await gateway.client.channels.get(d.channel_id)
+  // Fetch the channel if not cached
+  if(!channel) channel = (await gateway.client.channels.fetch(d.channel_id) as any) as Channel
```

Comment:
channels from event `messageCreate` will always be `TextChannel`

Reply:
Fixed.